src/Avalonia.Controls.TreeDataGrid/Models/TreeDataGrid/HierarchicalRows.cs

```diff
@@ -141,6 +141,19 @@ public void UnrealizeCell(ICell cell, int rowIndex, int columnIndex)
             (cell as IDisposable)?.Dispose();
         }
 
+        public int GetParentRowIndex(IndexPath modelIndex)
+        {
+            var index = ModelIndexToRowIndex(modelIndex);
+            for (int i = index - 1; i < index; i--)
```

Comment:
Is the `i < index` check correct here? `i` starts at `index - 1` and is decreased, so i think this is checking for when `i` wraps from `int.MinValue` to `int.MaxValue`?

Reply:
In addition, I don't think you need to do this iteration. Can't we just get the parent model index from `modelIndex` and then get the row index from that?